app/src/main/kotlin/com/wire/android/config/DefaultServerConfig.kt

```diff
@@ -18,18 +18,58 @@
 package com.wire.android.config
 
 import com.wire.android.BuildConfig
+import com.wire.android.emm.ManagedServerConfig
 import com.wire.kalium.logic.configuration.server.ServerConfig
+import javax.inject.Inject
+import javax.inject.Singleton
 
-val DefaultServerConfig = ServerConfig.Links(
-    api = BuildConfig.DEFAULT_BACKEND_URL_BASE_API,
-    accounts = BuildConfig.DEFAULT_BACKEND_URL_ACCOUNTS,
-    webSocket = BuildConfig.DEFAULT_BACKEND_URL_BASE_WEBSOCKET,
-    teams = BuildConfig.DEFAULT_BACKEND_URL_TEAM_MANAGEMENT,
-    blackList = BuildConfig.DEFAULT_BACKEND_URL_BLACKLIST,
-    website = BuildConfig.DEFAULT_BACKEND_URL_WEBSITE,
-    title = BuildConfig.DEFAULT_BACKEND_TITLE,
-    isOnPremises = false,
-    apiProxy = null
-)
+@Singleton
+class ServerConfigProvider @Inject constructor() {
+
+    fun getDefaultServerConfig(managedServerConfig: ManagedServerConfig? = null): ServerConfig.Links {
+        return if (managedServerConfig != null) {
+            with(managedServerConfig) {
+                ServerConfig.Links(
+                    api = endpoints.backendURL,
+                    accounts = endpoints.accountsURL,
+                    webSocket = endpoints.backendWSURL,
+                    teams = endpoints.teamsURL,
+                    blackList = endpoints.blackListURL,
+                    website = endpoints.websiteURL,
+                    title = title,
+                    isOnPremises = true, // EMM configuration always treated as on-premises
+                    apiProxy = null
+                )
+            }
+        } else {
+            ServerConfig.Links(
+                api = BuildConfig.DEFAULT_BACKEND_URL_BASE_API,
+                accounts = BuildConfig.DEFAULT_BACKEND_URL_ACCOUNTS,
+                webSocket = BuildConfig.DEFAULT_BACKEND_URL_BASE_WEBSOCKET,
+                teams = BuildConfig.DEFAULT_BACKEND_URL_TEAM_MANAGEMENT,
+                blackList = BuildConfig.DEFAULT_BACKEND_URL_BLACKLIST,
+                website = BuildConfig.DEFAULT_BACKEND_URL_WEBSITE,
+                title = BuildConfig.DEFAULT_BACKEND_TITLE,
+                isOnPremises = false,
+                apiProxy = null
+            )
+        }
+    }
+}
+
+// Keep the function for backward compatibility and Hilt module usage
+fun getDefaultServerConfig(managedServerConfig: ManagedServerConfig? = null): ServerConfig.Links {
+    val provider = ServerConfigProvider()
+    return provider.getDefaultServerConfig(managedServerConfig)
+}
```

Comment:
this makes me question how the domain registration should work?
should it use the baked in wire domain or the one from MDM ?

Reply:
The path won't go through wire domain, since this is the implementation in place for enterprise login, it will go to whatever it has on the deeplink configs, therefore if no domain registration endpoint is available in the BE, it will behave as any other on premises case.